src/register.ts

```diff
@@ -11,62 +9,14 @@ const isComponentRegistered = (name: string) => {
   return true;
 };
 
-const generateUniqueId = (length: number = 32): string => {
-  const possible = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
-
-  let text = '';
-  for (let i = 0; i < length; i++) {
-    text += possible.charAt(Math.floor(Math.random() * possible.length));
-  }
-
-  if (ELEMENT_IDS_BEING_USED.indexOf(text) < 0) {
-    ELEMENT_IDS_BEING_USED.push(text);
-    return text;
-  }
-  return generateUniqueId(length);
-}
-
-const isNotChild = (element: HTMLElement, _: number, elements: HTMLElement[]) => !elements
-  .filter(node => element !== node)
-  .reduce((result, node) => result || node.contains(element), false);
-
-interface UnregisteredHTMLElement {
-  oldElement: HTMLElement,
-  outerHTML: string;
-  id: string;
-}
-
 export const register = (component: typeof Component) => {
-  const dashedName = getComponentName(component);
-  const isRegistered = isComponentRegistered(dashedName);
-  let elements: UnregisteredHTMLElement[] = [];
-
-  // if component is not registered, remove any existing element from the page
-  if (!isRegistered) {
-    elements = ([].slice.call(document.getElementsByTagName(dashedName)) as HTMLElement[])
-    .filter(isNotChild)
-    .map(element => ({
-      oldElement: element,
-      outerHTML: element.outerHTML,
-      id: generateUniqueId(),
-    }));
-
-    elements.forEach(({ oldElement, id }) => oldElement.outerHTML = `<div id="${id}"></div>`);
+  if (!component) {
+    throw new Error('No component to register defined!')
   }
-
-  component.dependencies.forEach(dependency => dependency.register());
-
-  if (!isRegistered) {
-    component.define(dashedName);
-
-    // re-apply the previously removed elements from the page
-    elements.forEach(({ outerHTML, id }) => {
-      const element = document.getElementById(id);
-      const wrappingElement = document.createElement('div');
-      wrappingElement.innerHTML = outerHTML;
-
-      element.parentNode.replaceChild(wrappingElement.children[0], element);
-      ELEMENT_IDS_BEING_USED.splice(ELEMENT_IDS_BEING_USED.indexOf(id), 1);
-    })
+  const dashedName = getComponentName(component);
+  if (isComponentRegistered(dashedName)) {
+    throw new Error('Component already registered!')
```

Comment:
Should it just skip registering it instead? IMO it's a bad idea to throw an error if you, for example, import a component twice.

Reply:
Well at least it should notify the user. Then maybe a console warn? I just imaging a situation: 2 different developers build 2 different x-button components.
If this silently fails it will be hell to debug

Reply:
Then I'd be happy with suggest something like this:
```typescript
console.warn(`Re-registering component "${componentName}".`);
```

Reply:
Updated